web/components/search/query-users.ts

```diff
@@ -0,0 +1,18 @@
+import { User } from 'common/user'
+import { useEffect, useState } from 'react'
+import { searchUsers } from 'web/lib/supabase/users'
+
+// use nonce to make sure only latest result gets used
+let nonce = 0
+
+export function useUserSearchResults(query: string) {
+  const [results, setResults] = useState([] as User[])
+  useEffect(() => {
+    ++nonce
+    const thisNonce = nonce
+    searchUsers(query, 2).then((users) => {
+      if (thisNonce === nonce) setResults(users)
+    })
```

Comment:
I note that it would be reasonable to have a cache in these hooks of previously-fetched results per query, to make backspacing really fast, but searching is pretty fast anyway so I wouldn't say it's mandatory.